src/sonic-bgpcfgd/staticroutebfd/main.py

```diff
@@ -502,18 +502,26 @@ def static_route_del_handler(self, key, redis_del):
         arg_list    = lambda v: [x.strip() for x in v.split(',')] if len(v.strip()) != 0 else None
         nh_list     = arg_list(data['nexthop']) if 'nexthop' in data else None
         nh_vrf_list = arg_list(data['nexthop-vrf']) if 'nexthop-vrf' in data else None
-        for index in range(len(nh_list)):
-            nh_ip = nh_list[index]
-            nh_vrf = nh_vrf_list[index]
-            nh_key = nh_vrf + "|" + nh_ip
-            self.remove_from_nh_table_entry(nh_key, route_cfg_key)
+        bfd_field   = arg_list(data['bfd']) if 'bfd' in data else ["false"]
+        bfd_enabled = self.isFieldTrue(bfd_field)
+
+        # for a bfd_enabled static route, the nh_vrf_list was processed, has same length with nh_list
+        if bfd_enabled and nh_list and nh_vrf_list and len(nh_list) == len(nh_vrf_list):
+            for index in range(len(nh_list)):
+                nh_ip = nh_list[index]
+                nh_vrf = nh_vrf_list[index]
+                nh_key = nh_vrf + "|" + nh_ip
+                self.remove_from_nh_table_entry(nh_key, route_cfg_key)
+
+                if len(self.get_local_db(LOCAL_NEXTHOP_TABLE, nh_key)) == 0:
+                    bfd_key = nh_vrf + ":default:" + nh_ip
+                    self.remove_from_local_db(LOCAL_BFD_TABLE, bfd_key)
+                    self.del_bfd_session_from_appl_db(bfd_key)
 
-            if len(self.get_local_db(LOCAL_NEXTHOP_TABLE, nh_key)) == 0:
-                bfd_key = nh_vrf + ":default:" + nh_ip
-                self.remove_from_local_db(LOCAL_BFD_TABLE, bfd_key)
-                self.del_bfd_session_from_appl_db(bfd_key)
+        # do not delete it from appl_db if the route is not bfd enabled
+        if bfd_enabled:
```

Comment:
Why aren't we removing the static route if bfd not enabled? What was the previous behavior? Does this mean if the user removes static route via CLI, won't it get removed from APP_DB?

Reply:
If BFD is not enabled, staticroutebfd does not write static route entry to appl_db. so it should not delete it from appl_db. In general, prefix in config_db should not in appl_db. but to be safe, it is better that staticroutebfd does not touch that route in appl_db if it is not created by  staticroutebfd.